packages/contracts/contracts/test/L2OutputOracle.t.sol

```diff
@@ -565,3 +565,52 @@ contract L2OutputOracleUpgradeable_Test is L2OutputOracle_Initializer {
         assertEq(slot21Expected, slot21After);
     }
 }
+
+contract L2OutputOracle_Restricted_Test is ValidatorSystemUpgrade_Initializer {
+    function setUp() public virtual override {
+        super.setUp();
+
+        vm.prank(trusted);
+        pool.deposit{ value: trusted.balance }();
+        _registerValidator(trusted, minActivateAmount);
+
+        // Submit outputs to ValidatorPool is terminated
+        for (uint256 i; i <= terminateOutputIndex; i++) {
+            _submitL2OutputV1();
+        }
+    }
+
+    /******************************************
+     * Submit Tests - Restricted output index *
+     ******************************************/
+
+    function test_submitL2Output_restricted_trustedValidator_succeeds() public {
+        uint256 nextBlockNumber = oracle.nextBlockNumber();
+        uint256 nextOutputIndex = oracle.nextOutputIndex();
+        bytes32 dummyOutputRoot = keccak256(abi.encode());
+
+        vm.prank(trusted);
+        valMgr.setRestrictedOutputIndex(nextOutputIndex);
+        assertEq(valMgr.restrictedOutputIndex(), nextOutputIndex);
+
+        warpToSubmitTime();
```

Comment:
It's better to warp to public round, since trusted validator is always selected for the priority validator

```suggestion
        // Warp to public round
        vm.warp(oracle.nextOutputMinL2Timestamp() + roundDuration + 1);
        assertEq(valMgr.nextValidator(), trusted);

```